central/cluster/datastore/telemetry.go

```diff
@@ -76,47 +83,60 @@ var Gather phonehome.GatherFunc = func(ctx context.Context) (map[string]any, err
-	if cfg := centralclient.InstanceConfig(); cfg.Enabled() {
-		ctx = sac.WithGlobalAccessScopeChecker(ctx,
-			sac.AllowFixedScopes(
-				sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
-				sac.ResourceScopeKeys(resources.Cluster)))
-
-		cluster, ok, err := Singleton().GetCluster(ctx, clusterID)
-		if err != nil || !ok {
-			return
-		}
-		props := makeClusterProperties(cluster)
-		props["Total Nodes"] = metrics.NodeCount
-		props["CPU Capacity"] = metrics.CpuCapacity
-
-		if pmd := cluster.GetStatus().GetProviderMetadata(); pmd.GetProvider() != nil {
-			switch pmd.GetProvider().(type) {
-			case *storage.ProviderMetadata_Aws:
-				props["Provider"] = "AWS"
-			case *storage.ProviderMetadata_Azure:
-				props["Provider"] = "Azure"
-			case *storage.ProviderMetadata_Google:
-				props["Provider"] = "Google"
-			default:
-				props["Provider"] = "Unknown"
-			}
-			props["Provider Region"] = pmd.GetRegion()
-			props["Provider Zone"] = pmd.GetZone()
-			props["Provider Verified"] = pmd.GetVerified()
-		}
+	cfg := centralclient.InstanceConfig()
+	if !cfg.Enabled() {
+		return
+	}
 
-		omd := cluster.GetStatus().GetOrchestratorMetadata()
-		if omd.GetIsOpenshift() != nil {
-			props["Openshift"] = omd.GetIsOpenshift()
-		}
-		props["Orchestrator Version"] = omd.GetVersion()
+	ctx = sac.WithGlobalAccessScopeChecker(ctx,
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Cluster)))
 
-		opts := []telemeter.Option{
-			telemeter.WithClient(cluster.GetId(), securedClusterClient),
-			telemeter.WithGroups(cfg.GroupType, cfg.GroupID),
-			telemeter.WithTraits(props),
+	cluster, ok, err := Singleton().GetCluster(ctx, clusterID)
+	if err != nil || !ok {
+		return
+	}
+	props := makeClusterProperties(cluster)
+	props["Total Nodes"] = metrics.NodeCount
+	props["CPU Capacity"] = metrics.CpuCapacity
+
+	if pmd := cluster.GetStatus().GetProviderMetadata(); pmd.GetProvider() != nil {
+		switch pmd.GetProvider().(type) {
+		case *storage.ProviderMetadata_Aws:
+			props["Provider"] = "AWS"
+		case *storage.ProviderMetadata_Azure:
+			props["Provider"] = "Azure"
+		case *storage.ProviderMetadata_Google:
+			props["Provider"] = "Google"
+		default:
+			props["Provider"] = "Unknown"
 		}
-		cfg.Telemeter().Track("Updated Secured Cluster Identity", nil, opts...)
+		props["Provider Region"] = pmd.GetRegion()
+		props["Provider Zone"] = pmd.GetZone()
+		props["Provider Verified"] = pmd.GetVerified()
+	}
+
+	omd := cluster.GetStatus().GetOrchestratorMetadata()
+	if omd.GetIsOpenshift() != nil {
+		props["Openshift"] = omd.GetIsOpenshift()
+	}
+	props["Orchestrator Version"] = omd.GetVersion()
+
+	sendProps(cfg, cluster, props)
+}
+
+func sendProps(cfg *phonehome.Config, cluster *storage.Cluster, props map[string]any) {
+	clusterID := cluster.GetId()
+	if lastProps, ok := clusterIdentityCache.Get(clusterID); ok && reflect.DeepEqual(lastProps, props) {
```

Comment:
if we have a `map[string]string` then we can use https://pkg.go.dev/golang.org/x/exp/maps#Equal

Reply:
There's also "github.com/mitchellh/hashstructure/v2" for random types.